src/main/java/com/faforever/client/vault/replay/ReplayVaultController.java

```diff
@@ -129,6 +137,30 @@ protected void loadLocalReplaysInBackground() {
     return row;
   }
 
+  @NotNull
+  private ContextMenu replayRowContextMenuFactory(Replay replay) {
+    ContextMenu contextMenu = new ContextMenu();
+    MenuItem copyReplayId = new MenuItem(i18n.get("vault.replays.copyReplayId"));
+    copyReplayId.setOnAction(event -> {
+      int replayId = replay.getId();
+      final Clipboard clipboard = Clipboard.getSystemClipboard();
+      final ClipboardContent content = new ClipboardContent();
+      content.putString(String.valueOf(replayId));
+      clipboard.setContent(content);
+    });
+    MenuItem copyReplayPath = new MenuItem(i18n.get("vault.replays.copyReplayPath"));
+    copyReplayPath.setOnAction(event -> {
+      Path replayPath = replay.getReplayFile();
+      final Clipboard clipboard = Clipboard.getSystemClipboard();
+      final ClipboardContent content = new ClipboardContent();
+      content.putString(replayPath.toFile().getAbsolutePath());
+      clipboard.setContent(content);
```

Comment:
Duplicate code. I don't know if we use clipboard features elsewhere, but if yes a clipboard util class probably makes sense, otherwise at least this should be extracted into a method here.

Reply:
I have created a `ClipboardUtil` class and used it also in `ChatUserContextMenuController`  and `ReplayDetailController`